api/v1alpha1/tenant_webhook.go

```diff
@@ -54,6 +54,16 @@ var _ webhook.Defaulter = &Tenant{}
 // Default implements webhook.Defaulter so a webhook will be registered for the type
 func (t *Tenant) Default() {
 	Log.Info("Validating default for", "tenant", t.Name)
+
+	if t.GetDeletionTimestamp() != nil {
+		t.Spec.State = "Deleting"
```

Comment:
State looks like a string masquerading as an enum, any reason not to do something like:

```
type State int64

const (
	New State = iota
	Deleting
...
)

func (s State) String() string {
	switch s {
	case New:
		return "new"
	case Deleting:
		return "deleting"
...
	}
	return "bug"
}
```

Then the State field can simply assign New/Deleting etc... as known constants and it can make validation a little easier.

Reply:
nice -- will include in future PRs..